examples/benchmark/tls_bench.c

```diff
@@ -605,34 +625,122 @@ static void print_stats(stats_t* wcStat, const char* desc, const char* cipher, i
            wcStat->connTime * 1000,
            wcStat->connTime * 1000 / wcStat->connCount);
 }
+
+static void Usage(void)
+{
+    printf("tls_bench "    LIBWOLFSSL_VERSION_STRING
+           " NOTE: All files relative to wolfSSL home dir\n");
+    printf("-?          Help, print this usage\n");
+    printf("-b <num>    The total <num> bytes transferred per test connection, default 1 MB\n");
+#ifdef DEBUG_WOLFSSL
+    printf("-d          Enable debug messages\n");
+#endif
+    printf("-e          List Every cipher suite available, \n");
+    printf("-l <str>    Cipher suite list (: delimited)\n");
+    printf("-t <num>    Time <num> (seconds) to run each test, default 1\n");
+    printf("-p <num>    The packet size <num> in bytes [1-16kB], default 1 kB\n");
+}
+
+static void ShowCiphers(void)
+{
+    char ciphers[4096];
+
+    int ret = wolfSSL_get_ciphers(ciphers, (int)sizeof(ciphers));
+
+    if (ret == WOLFSSL_SUCCESS)
+        printf("%s\n", ciphers);
+}
+
 #ifdef __GNUC__
 #pragma GCC diagnostic pop
 #endif
 
-int bench_tls(void)
+int bench_tls(void* args)
 {
     info_t theadInfo[THREAD_COUNT];
     info_t* info;
     int i, doShutdown;
     char *cipher, *next_cipher, ciphers[4096];
+    int     argc = ((func_args*)args)->argc;
+    char**  argv = ((func_args*)args)->argv;
+    int    ch;
 
-#ifdef DEBUG_WOLFSSL
-    wolfSSL_Debugging_ON();
-#endif
+
+    /* Vars configured by command line arguments */
+    int argRuntimeSec = 1;
+    char *argCipherList = NULL;
+    int argTestSizeBytes = (1024 * 1024);
+    int argTestPacketSize = 1024;
+
+    /* Avoid unused warnings */
+    (void) argRuntimeSec;
+    (void) argCipherList;
+    (void) argTestSizeBytes;
+    (void) argTestPacketSize;
+
+    ((func_args*)args)->return_code = -1; /* error state */
 
     /* Initialize wolfSSL */
     wolfSSL_Init();
 
-    /* Run for each cipher */
-    wolfSSL_get_ciphers(ciphers, (int)sizeof(ciphers));
+    /* Parse command line arguments */
+    while ((ch = mygetopt(argc, argv, "?" "b:del:p:t:")) != -1) {
+        switch (ch) {
+            case '?' :
+                Usage();
+                exit(EXIT_SUCCESS);
+
+            case 'b' :
+                argTestSizeBytes = atoi(myoptarg);
+                break;
+
+#ifdef DEBUG_WOLFSSL
+            case 'd' :
```

Comment:
Can you add the `-d` information to the usage help when DEBUG_WOLFSSL is defined (`./configure --enable-debug`)?

Also, can you add runtime argument options for `SHOW_PEER_INFO` as `-i` and `SHOW_VERBOSE_OUTPUT` as `-v`?